test/cpp/api/nn_utils.cpp

```diff
@@ -79,7 +79,7 @@ TEST_F(NNUtilsTest, ClipGradNorm) {
   };
   for (auto norm_type : norm_types) {
     for (int i = 0; i < grads.size(); i++) {
-      l->parameters()[i].grad().data().copy_(grads[i]);
+      l->parameters()[i].mutable_grad().data().copy_(grads[i]);
```

Comment:
This one too.

Reply:
Ho the .data() removed the const-ness :'( good to know that it is as bad in cpp as it is in python haha